zstd/seqdec.go

```diff
@@ -181,11 +181,13 @@ func (s *sequenceDecs) decode(seqs int, br *bitReader, hist []byte) error {
 			return fmt.Errorf("output (%d) bigger than max block size", size)
 		}
 		if size > cap(s.out) {
-			// Not enough size, will be extremely rarely triggered,
+			// Not enough size, which can happen under high volume block streaming conditions
 			// but could be if destination slice is too small for sync operations.
-			// We add maxBlockSize to the capacity.
-			s.out = append(s.out, make([]byte, maxBlockSize)...)
-			s.out = s.out[:len(s.out)-maxBlockSize]
+			// over-allocating here can create a large amount of GC pressure so we try to keep
+			// it as contained as possible
+			addBytes := 1024 + (size - len(s.out))
```

Comment:
```suggestion
			used := len(s.out) - startSize
			addBytes := 256 + ll + ml + used>>2

			// Clamp to max block size.
			if used+addBytes > maxBlockSize {
				addBytes = maxBlockSize - used
			}
```

Reply:
Maybe something like this?

Reply:
ok gave that change a whirl
there's about a 500-750Mb difference in the ram usage, higher using method, then the original `size - len` one, but it does seem to not get out of control, which is the desired effect.

Reply:
Great. I will merge when Ci tests are done and probably release it tomorrow.